lib/node-utils.js

```diff
@@ -0,0 +1,634 @@
+TOKEN_TO_TEXT[SyntaxKind.BarBarToken] = "||";
+TOKEN_TO_TEXT[SyntaxKind.QuestionToken] = "?";
+TOKEN_TO_TEXT[SyntaxKind.ColonToken] = ":";
+TOKEN_TO_TEXT[SyntaxKind.EqualsToken] = "=";
+TOKEN_TO_TEXT[SyntaxKind.PlusEqualsToken] = "+=";
+TOKEN_TO_TEXT[SyntaxKind.MinusEqualsToken] = "-=";
+TOKEN_TO_TEXT[SyntaxKind.AsteriskEqualsToken] = "*=";
+TOKEN_TO_TEXT[SyntaxKind.SlashEqualsToken] = "/=";
+TOKEN_TO_TEXT[SyntaxKind.PercentEqualsToken] = "%=";
+TOKEN_TO_TEXT[SyntaxKind.LessThanLessThanEqualsToken] = "<<=";
+TOKEN_TO_TEXT[SyntaxKind.GreaterThanGreaterThanEqualsToken] = ">>=";
+TOKEN_TO_TEXT[SyntaxKind.GreaterThanGreaterThanGreaterThanEqualsToken] = ">>>=";
+TOKEN_TO_TEXT[SyntaxKind.AmpersandEqualsToken] = "&=";
+TOKEN_TO_TEXT[SyntaxKind.BarEqualsToken] = "|=";
+TOKEN_TO_TEXT[SyntaxKind.CaretEqualsToken] = "^=";
+TOKEN_TO_TEXT[SyntaxKind.AtToken] = "@";
+TOKEN_TO_TEXT[SyntaxKind.InKeyword] = "in";
+
+/**
+ * Find the first matching child based on the given sourceFile and predicate function.
+ * @param {TSNode} node The current TSNode
+ * @param {Object} sourceFile The full AST source file
+ * @param {Function} predicate The predicate function to apply to each checked child
+ * @returns {TSNode|undefined} a matching child TSNode
+ */
+function findFirstMatchingChild(node, sourceFile, predicate) {
+    const children = node.getChildren(sourceFile);
+    for (let i = 0; i < children.length; i++) {
+        const child = children[i];
+        if (child && predicate(child)) {
+            return child;
+        }
+    }
+    return undefined
+}
+
+/**
+ * Find the first matching ancestor based on the given predicate function.
+ * @param {TSNode} node The current TSNode
+ * @param {Function} predicate The predicate function to apply to each checked ancestor
+ * @returns {TSNode|undefined} a matching parent TSNode
+ */
+function findFirstMatchingAncestor(node, predicate) {
+    while (node) {
+        if (predicate(node)) {
+            return node;
+        }
+        node = node.parent;
+    }
+    return undefined;
+}
+
+/**
+ * Returns true if the given TSNode is a let variable declaration
+ * @param {TSNode} node The TSNode
+ * @returns {boolean} whether or not the given node is a let variable declaration
+ */
+function isLet(node) {
+    /**
+     * TODO: Remove dependency on private TypeScript method
+     */
+    return ts.isLet(node);
+}
+
+/**
+ * Returns true if the given TSNode is a const variable declaration
+ * @param {TSNode} node The TSNode
+ * @returns {boolean} whether or not the given node is a const variable declaration
+ */
+function isConst(node) {
+    /**
+     * TODO: Remove dependency on private TypeScript method
+     */
+    return ts.isConst(node);
+}
+
+//------------------------------------------------------------------------------
+// Public
+//------------------------------------------------------------------------------
+
+/**
+ * Expose the enum of possible TSNode `kind`s.
+ */
+exports.SyntaxKind = SyntaxKind;
```

Comment:
Nit: Maybe we should move the exports to the bottom of the file and do it all in one object. That way it might be easier to know what a file exports. If this was typescript we could just export the function or variables but I don't care too much.